pkg/storage/internalstorage/util.go

```diff
@@ -158,7 +158,13 @@ func applyListOptionsToQuery(query *gorm.DB, opts *internal.ListOptions, applyFn
 	var amount *int64
 	if opts.WithRemainingCount != nil && *opts.WithRemainingCount {
 		amount = new(int64)
-		query = query.Count(amount)
+		countQuery := *query
+		statement := *countQuery.Statement
+		countQuery.Statement = &statement
+		if len(countQuery.Statement.Selects) == 1 && countQuery.Statement.Selects[0] == objectColumn {
+			countQuery.Statement.Selects = []string{"*"}
+		}
+		countQuery.Count(amount)
```

Comment:
This may seem like a very hack way of doing things, but in reality we want to skip the field check when `Count` and just put the select at the end.

https://github.com/clusterpedia-io/clusterpedia/blob/1d4658d1c4f3c2f815d335f7a79a3d7d5f39f3e5/pkg/storage/internalstorage/resource_storage.go#L199-L200
->
```go
        if err := result.From(result.Select(query)); err != nil {
                return InterpretDBError(s.storageGroupResource.String(), err)
        }
```